extension-manifest-v3/src/background/stats.js

```diff
@@ -57,11 +58,18 @@ async function refreshIcon(tabId) {
         stats.trackers.map((t) => t.category),
       );
     }
-    try {
-      await chrome.action.setIcon({ tabId, ...data });
-    } catch (e) {
-      console.error('Error while trying update the icon:', e);
-    }
+
+    // Note: Even in MV3, this is not (yet) returning a promise.
+    chrome.action.setIcon({ tabId, ...data }, () => {
```

Comment:
should we still await it? how about using `webextension-polyfill`? 

Below there is a call to `setBadgeText` - doesn't it have the same problem?